src/tile-board/Tileboard.jsx

```diff
@@ -36,4 +48,34 @@ function createArrayGrid() {
   return allTiles;
 }
 
+/**
+ * Toggles adjacent tiles
+ */
+function toggle(arrayGrid, id, size) {
+  const tempArray = [...arrayGrid];
```

Comment:
Since `arrayGrid` is a nested array, a shallow copy will retain references to the inner array elements leading to a mutation below when you assign to `tempArray[i][j]`. Mutating data that lives in a react `useState` variable can lead to unexpected rendering behavior. Can you think of another way to structure this so that you avoid the mutation altogether?